File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.cpp
class Solution {
public:
    long long maxMatrixSum(vector<vector<int>>& matrix) {
        long long ans=0;
        int min_val=INT_MAX;
        bool odd_neg=0;
        for(auto row:matrix){
            for(auto element:row){
                ans+=abs(element);
                if(element<0)
                    odd_neg^=1;
                min_val=min(min_val,abs(element));
            }
        }
        ans-=2*odd_neg*min_val;
        return ans;
    }
};

File: 1975-maximum-matrix-sum/1975-maximum-matrix-sum.py
class Solution:
    def maxMatrixSum(self, matrix: List[List[int]]) -> int:
        mn = 10**5
        summ = 0
        isNegOdd = False
        for row in matrix:
            for num in row:
                summ += abs(num)
                mn = min(mn, abs(num))
                if num < 0:
                    isNegOdd = 1 - isNegOdd
        if isNegOdd:
            summ -= 2 * mn
        return summ
